src/main/java/sirius/biz/jobs/params/BooleanParameter.java

```diff
@@ -47,18 +50,41 @@ protected String checkAndTransformValue(Value input) {
         if (isRequired() && !input.asBoolean()) {
             return null;
         }
-
+        if (nullable && !resolveFromString(input).isPresent()) {
+            return null;
+        }
         return NLS.toMachineString(input.asBoolean());
     }
 
+    /**
+     * Marks the parameter as boolean.
+     * <p>
+     * This allows for a tri-state boolean where a job parameter can also receive
+     * {@code null} when a value is not selected instead of being defaulted to false.
+     *
+     * @return the parameter itself for fluent method calls
+     */
+    public BooleanParameter markNullable() {
+        this.nullable = true;
+        return self();
+    }
+
     @Override
     public BooleanParameter markRequired() {
         throw new UnsupportedOperationException(
-                "A boolean parameter must not be marked as required as it is inherently so!");
+                "A boolean parameter must not be marked as required as it is inherently so. Use markNullable() to make the parameter optional");
     }
 
     @Override
-    protected Optional<Boolean> resolveFromString(Value input) {
+    public boolean isRequired() {
+        return !nullable;
+    }
+
+    @Override
+    protected Optional<Boolean> resolveFromString(@Nonnull Value input) {
+        if (nullable && !("true".equalsIgnoreCase(input.asString()) || "false".equalsIgnoreCase(input.asString()))) {
```

Comment:
this does not match the parsing of input.asBoolean which is used when the parameter is required, as $NLS.yes is also considerd as true (and $NLS.no as false)
->
if the parameter is required i will get true for "yes"
if the parameter is required i will get null for "yes"

Reply:
my suggestion:
```suggestion
        if (nullable && !input.isFilled()) {
```

Reply:
Cannot use input.asBoolean here (is actually the whole point of it), because it evaluates anything else as false. In this usage case (and the actual values of the selectBoolean template) only allow (and expect) true or false.

Reply:
but "yes" and "no" should still be parsed as booleans, and null should possibly only be returned if the value is not filled at all?